encoding/basics-and-workflow.md

```diff
@@ -2,36 +2,584 @@
+
+```
+458.233.964 KaguyaBD_01_(original).wav
+314.660.781 KaguyaBD_01_(00).flac
+312.773.683 KaguyaBD_01_(01).flac
+312.486.815 KaguyaBD_01_(02).flac
+300.712.094 KaguyaBD_01_(03).flac
+299.806.321 KaguyaBD_01_(04).flac
+299.658.920 KaguyaBD_01_(05).flac
+299.606.279 KaguyaBD_01_(06).flac
+300.119.565 KaguyaBD_01_(07).flac
+298.832.315 KaguyaBD_01_(08).flac
+299.172.074 KaguyaBD_01_(09).flac
+298.773.829 KaguyaBD_01_(10).flac
+298.468.836 KaguyaBD_01_(11).flac
+297.952.201 KaguyaBD_01_(12).flac
+```
+
+We can drastically reduce the filesize
+by making use of `-sample_fmt s16`.
+Not a lot of anime have true 24bit audio,
+and the vast majority of people
+wouldn't be able to tell a difference regardless.
+Some encoders prefer encoding FLAC in 16bit instead
+for this very reason.
+
+```
+458.233.964 KaguyaBD_01_(original).wav
+163.088.307 KaguyaBD_01_(00)_16bit.flac
+161.313.003 KaguyaBD_01_(01)_16bit.flac
+161.051.956 KaguyaBD_01_(02)_16bit.flac
+149.929.246 KaguyaBD_01_(03)_16bit.flac
+149.084.733 KaguyaBD_01_(04)_16bit.flac
+148.909.370 KaguyaBD_01_(05)_16bit.flac
+148.857.694 KaguyaBD_01_(06)_16bit.flac
+148.848.250 KaguyaBD_01_(07)_16bit.flac
+148.149.100 KaguyaBD_01_(08)_16bit.flac
+148.088.813 KaguyaBD_01_(09)_16bit.flac
+148.080.093 KaguyaBD_01_(10)_16bit.flac
+147.368.127 KaguyaBD_01_(11)_16bit.flac
+147.327.448 KaguyaBD_01_(12)_16bit.flac
+```
+
+AAC serves as a fantastic middleground
+between FLAC and opus.
+AAC is a lossy audio codec,
+which results in overall smaller encodes
+at the cost of it throwing away some data.
+
+For encoding in AAC it is always recommended
+to use [qaac](qaac) (the Apple AAC encoder)
+over the AAC encoder supplied with ffmpeg.
+It does a much better job of preserving the perceptional quality
+over all its competitors,
+and is thus the go-to for people that encode in AAC.
+
+The options you'll used most is `-V`.
+`-V` sets the true Variable Bitrate mode,
+allowing qaac free reign in deciding
+what it believes to be the best bitrate to encode to.
+
+The most common command looks like this:
+
+`$ qaac input.wav -V 127`
+
+Let's compare the filesizes between the FLAC encodes we did before.
+
+```
+26.811.464 Fate_Zero_OP2_(original).wav
+19.469.016 Fate_Zero_OP2_(12)_24bit.flac
+10.898.384 Fate_Zero_OP2_(12)_16bit.flac
+ 3.886.831 Fate_Zero_OP2_(qaac).m4a
+```
+```
+458.233.964 KaguyaBD_01_(original).wav
+297.952.201 KaguyaBD_01_(12)_24bit.flac
+147.327.448 KaguyaBD_01_(12)_16bit.flac
+ 54.191.826 KaguyaBD_01_(qaac).m4a
+```
+
+The final commonly-used audio codec is opus.
+Opus is used for web audio,
+as it is designed for
+low-latency, low-complexity audio compression.
+This makes it great for "mini" releases.
+You can encode to qaac through ffmpeg:
+
+`$ ffmpeg -i input.wav output.opus`
+
+Or through [Opus-tools](opus-tools)
+if you don't know
+how to compile ffmpeg with opus support:
+
+`$ opusenc input.wav output.opus`
+
+Unlike qaac, opus works with a constant bitrate.
```

Comment:
Opus also has vbr.

Reply:
Fwiw, FFmpeg's libopus uses vbr by default, and it is generally [reccomended](https://mf4.xiph.org/jenkins/view/opus/job/opus-tools/ws/man/opusenc.html#--bitrate) over cbr and cvbr. The `-b:a` option for libopus uses that number as a vbr estimate by default.

Reply:
Opus uses vbr by default.